iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/analyzer/StatementAnalyzer.java

```diff
@@ -650,16 +649,16 @@ protected Scope visitInsert(Insert insert, Optional<Scope> scope) {
               .collect(toImmutableList());
       analysis.registerTable(insert.getTable(), tableSchema, targetTable);
 
-      List<String> tableColumns =
-          columns.stream().map(ColumnSchema::getName).collect(toImmutableList());
+      Set<String> tableColumns =
+          columns.stream().map(ColumnSchema::getName).collect(toImmutableSet());
 
-      List<String> insertColumns;
+      Set<String> insertColumns;
```

Comment:
I think we need to keep the order of insertColumns? so maybe `LinkedHashSet` will be the best choice?